metrics/firebase/functions/lib/main.dart

```diff
@@ -61,47 +137,93 @@ int _getSuccessfulBuildDuration(BuildData buildData) {
 /// from the given [dateTime].
 DateTime _getUtcDay(DateTime dateTime) => dateTime.toUtc().date;
 
+/// Creates an instance of the [BuildDayStatusField] based on
+/// the given [buildStatus] and [incrementCount].
+BuildDayStatusField _createBuildDayStatusField({
+  BuildStatus buildStatus,
+  int incrementCount,
+}) {
+  const statusFieldMapper = BuildDayStatusFieldNameMapper();
+
+  final buildDayStatusFieldName = statusFieldMapper.map(buildStatus);
+
+  return BuildDayStatusField(
+    name: buildDayStatusFieldName,
+    value: Firestore.fieldValues.increment(incrementCount),
+  );
+}
+
 /// Updates the build day's document data with the given [buildDayData].
 ///
 /// Adds a new document to the tasks collection if updating
 /// the build day's document data fails.
-Future<void> _updateBuildDay(
-  DocumentSnapshot snapshot,
+///
+/// Throws an [ArgumentError] if either [buildDayData] or
+/// [onErrorData] is `null`.
+///
+/// Returns `true` if setting the given [buildDayData] to the build day
+/// document is successful. Otherwise, returns `false`.
+Future<bool> _updateBuildDay(
+  Firestore firestore, {
   BuildDayData buildDayData,
-) async {
+  TaskData onErrorData,
```

Comment:
```suggestion
  TaskData errorTaskData,
```
What do you think? Also, would it be great to update the documentation and mention this field in it?